tests/benchdnn/graph/graph.hpp

```diff
@@ -42,8 +42,8 @@ struct settings_t : public base_settings_t {
 
     // ctor to save certain fields from resetting
     settings_t(const char *perf_template) : settings_t() {
-        this->perf_template
-                = perf_template; // NOLINT(cppcoreguidelines-prefer-member-initializer)
+        // NOLINTNEXTLINE(cppcoreguidelines-prefer-member-initializer)
+        this->perf_template = perf_template;
     }
```

Comment:
Couldn't this be
```suggestion
    settings_t(const char *perf_template) : perf_template(perf_template) {}
```
and we can get rid of the suppression altogether?

Reply:
It's possible that this will have some unintended effect on value initialization, but I'll give it a shot and run graph tests to see.

Reply:
Please see the discussion here: https://github.com/uxlfoundation/oneDNN/pull/2742#discussion_r1968133318

Reply:
@TaoLv What do we gain from delegating to a default constructor? I _think_ that because all member variables have default values, they should pick those up (identically to what the default constructor would do) before overriding from `perf_template` from the member initializer list.

Reply:
All of the members of `settings_t` has a default initializer/constructor, so there is no need to explicitly call `settings_t()` here -- default initialization happens anyway. But delegating to another constructor forbids you from adding the `perf_template` initializer, as noted in the linked discussion.